manager/integration/tests/test_ha.py

```diff
@@ -2675,15 +2803,77 @@ def test_engine_image_not_fully_deployed_perform_volume_operations():
        returns error
     4. Verify that we can attach to another node, take snapshot, take a backup,
        expand, then detach vol-1
-    5. Verify that vol-2 cannot be attached to any nodes because one of
-       its replicas is sitting on the node-1 which doesn't have the
-       engine image. The attach API call returns error
+    5. Verify that vol-2 cannot be attached to tainted nodes. The attach API
+       call returns error
+    6. Verify that vol-2 can attach to non-tainted node with degrade status
     """
-    pass
+    volume1, volume2, tainted_node_id = \
+        prepare_engine_not_fully_deployed_evnironment_with_volumes(client,
+                                                                   core_api)
 
+    volume1 = client.by_id_volume(volume1.name)
+    volume1 = wait_for_volume_healthy(client, volume1.name)
 
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_engine_image_not_fully_deployed_perform_engine_upgrade():
+    # did not do data write because volume is not attached to self host node
```

Comment:
Suggest putting this `TODO`. If I remember correctly @yangchiu is working on making this doable.